src/Native/System.Security.Cryptography.Native/pal_hmac.cpp

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include "pal_config.h"
+#include "pal_utilities.h"
+#include "pal_hmac.h"
+
+#include <assert.h>
+#include <memory>
+#include <openssl/hmac.h>
+
+// Return values of certain HMAC functions vary per platform/OpenSSL
+// version. On some platforms, void is returned and success must be
+// assumed as there is no provision for failure. On others, int is
+// returned with 1 signifying success and 0 signifying failure.
+//
+// We call these functions indirectly via overloaded HmacCall helper
+// that will synthesize a successful result of 1 on platforms where
+// the functions return void and otherwise just forward along the
+// actual return value.
+
+template <typename F, typename... Args>
+inline static auto HmacCall(F func, Args... args) -> ReplaceVoidResultOf<F(Args...), int>
+{
+    func(args...);
+    return 1;
+}
+
+template <typename F, typename... Args>
+inline static auto HmacCall(F func, Args... args) -> NonVoidResultOf<F(Args...)>
+{
+    return func(args...);
+}
+
+extern "C" HMAC_CTX* HmacCreate(const uint8_t* key, int32_t keyLen, const EVP_MD* md)
+{
+    assert(key != nullptr || keyLen == 0);
+    assert(keyLen >= 0);
+    assert(md != nullptr);
+
+    std::unique_ptr<HMAC_CTX> ctx(new (std::nothrow) HMAC_CTX);
+    if (ctx == nullptr)
+    {
+        assert(false && "Allocation failed.");
+        return nullptr;
+    }
+
+    // !REVIEW! Zeroing out HMAC_CTX prior to init mimics prior managed initialization
+    //          and our tests fail without it, but I can't find a reference stating it's
+    //          necessary.
+    *ctx = {};
```

Comment:
You need to do both this -and- HMAC_CTX_init?  Because all HMAC_CTX_init does is zero out some fields; and HMAC_Init_ex resets the rest of them (I don't see it ever reading from the current structure to decide initialization already happened)